### PR TITLE
Implement project overview page

### PR DESCRIPTION
**Pull request by TheRealJon**
Basic project overview implementation with grouping and filtering of resources. Deployment Configs, Deployments, and Stateful Sets are currently the only top-level resources shown.

image

@openshift/team-ux-review

### CODE REVIEW

File: frontend/public/components/project-overview.jsx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+import { StartGuide } from './start-guide';
+import { FLAGS, connectToFlags, flagPending } from '../features';
+import { Firehose, NavTitle, ResourceName, SectionHeading } from './utils';
+import { DeploymentConfigOverviewList } from './deployment-config';
+
+const LIST_VIEW = 'list';
+const GRAPH_VIEW = 'graph';
+
+const ProjectOverviewToolbar = ({activeView, setView}) => {
+  return <Toolbar>
```
Comment (benjaminapetersen) [inline frontend/public/components/project-overview.jsx]: @sg00dwin ping!
I know you are consolidating a handful of toolbars throughout the console, would be good to take a quick look at this for future consistency.

File: frontend/public/components/project-overview.jsx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+          onClick={() => setView(GRAPH_VIEW)}
+          title="Topology Graph" >
+          <Icon type="pf" name="topology" />
+        </Button>
+      </Toolbar.ViewSelector>
+    </Toolbar.RightContent>
+  </Toolbar>;
+};
+
+const DeploymentConfigOverview = ({obj}) => {
```
Comment (benjaminapetersen) [inline frontend/public/components/project-overview.jsx]: I'm inclined to split this out early into separate component files, as well as see how generic they can be (perhaps smaller pieces will allow for generic lists with some nested child components for customization, perhaps imported from our existing `deployment-config.tsx`, etc files, not sure).  

Thinking about structure early for deduplication sake.

Comment (benjaminapetersen) [reply frontend/public/components/project-overview.jsx]: Ah, I see you are doing some of that already, nice.

Comment (TheRealJon) [reply frontend/public/components/project-overview.jsx]: Yes, plan on doing similar breaking down for other resources that will be displayed in this view, but just haven't gotten there yet. I wanted to put up a PR so that others could see what I am working on and start getting feedback early, but there's still work to be done.

File: frontend/package.json [hunk 3]
```diff
@@ -66,6 +66,7 @@
     "js-yaml": "3.x",
     "lodash-es": "4.x",
     "patternfly": "^3.45.0",
+    "patternfly-react": "^2.11.2",
```
Comment (jeff-phillips-18) [inline frontend/package.json]: We should go to 2.12, that will reduce the size cost of including pf-react since with that release it no longer pulls in all of react-bootstrap (only the components actually used).

File: frontend/public/components/project-overview.jsx [hunk 4]
```diff
@@ -0,0 +1,341 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as classNames from 'classnames';
+import { Button, EmptyState, Icon, Toolbar } from 'patternfly-react';
+import { Helmet } from 'react-helmet';
```
Comment (alecmerdler) [inline frontend/public/components/project-overview.jsx]: Newline between `node_modules` imports and project local imports for consistency.

File: frontend/package.json (hunk 3 above)
Comment (benjaminapetersen) [reply frontend/package.json]: +1 for not importing the whole world!

File: frontend/public/components/project-overview.jsx [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+          controller: true
+        });
+      });
+      const ordered = this.sortByDeploymentVersion(dcControllers, true);
+      deploymentConfig.replicationControllers = dcControllers;
+      deploymentConfig.currentController = _.head(ordered);
+      deploymentConfig.prevController = _.size(ordered) < 2 ? null : ordered[1];
+
+      // Get the services for this deployment config
+      let configServices = [];
```
Comment (benjaminapetersen) [inline frontend/public/components/project-overview.jsx]: Nit, I'd suggest a `const configServices = _.reduce(allServices, () => {}, [])` to make these lists.

**Comment by TheRealJon** [comment]
Added grouping functionality and adjusted styling according to style docs.

image

**Comment by TheRealJon** [comment]
@spadgett do we want to call this the first iteration? Or should I continue iterating on this PR to get the details panel working?

**Comment by spadgett** [comment]
@TheRealJon I'd suggest we integrate this in phases. Let's work to get what you have merged.

**Comment by spadgett** [comment]
jenkins rebuild

File: frontend/public/components/deployment-config.tsx [hunk 6]
```diff
@@ -47,52 +47,56 @@ const menuActions = [
   ...common,
 ];
 
-export const DeploymentConfigsDetails: React.SFC<{obj: any}> = ({obj: deploymentConfig}) => {
+export const DeploymentConfigOverviewList = ({deploymentConfig}) => {
```
Comment (spadgett) [inline frontend/public/components/deployment-config.tsx]: Let's pull this out of the PR for the moment and work on the details sidebar in a separate PR

File: frontend/public/components/overview.jsx [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+      controller: true
+    });
+  });
+};
+
+const OverviewToolbar = ({activeView, groupOptions, handleFilterChange, handleGroupChange, selectedGroup, setView, showViewSelector = false}) =>
+  <Toolbar className="overview-toolbar">
+    <Toolbar.RightContent>
+      <div className="form-group overview-toolbar__form-group">
+        <label style={{marginRight: '10px'}}>
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Avoid inline styles

File: frontend/public/components/overview.jsx [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+          Group by label
+        </label>
+        <Dropdown
+          className="overview-toolbar__dropdown"
+          disabled={!_.size(groupOptions)}
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]:
```
disabled={_.isEmpty(groupOptions)}
```

File: frontend/public/components/overview.jsx [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+          items={groupOptions}
+          onChange={handleGroupChange}
+          style={{display: 'inline-block'}}
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Avoid inline styles

File: frontend/public/components/overview.jsx [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+          title={selectedGroup}
+        />
+      </div>
+      <div className="form-group overview-toolbar__form-group">
+        <TextFilter
+          label="Resources by name"
+          onChange={handleFilterChange}
+        />
+      </div>
+      { showViewSelector &&
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: I would take this out of this PR. We can introduce it when we're ready to integrate topology view

File: frontend/public/components/overview.jsx [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
+    const compareDeployments = (left, right) => {
+      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
+      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
+      // Fall back to sorting by name if no deployment versions.
+      let leftName, rightName;
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: You can make this `const` and declare inside the `if`

File: frontend/public/components/overview.jsx [hunk 12]
```diff
@@ -0,0 +1,404 @@
...
+        return rightVersion - leftVersion;
+      }
+
+      return leftVersion - rightVersion;
+    };
+
+    return _.toArray(replicationControllers).sort(compareDeployments);
+  }
+
+  filterItems(items) {
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: We should be using `fuzzy` to filter things for consistency with other views.

File: frontend/public/components/overview.jsx [hunk 13]
```diff
@@ -0,0 +1,404 @@
...
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      const name = _.get(item, 'metadata.name', '');
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Names have to be lowercase, so we can avoid `toLowerCase`

File: frontend/public/components/overview.jsx [hunk 14]
```diff
@@ -0,0 +1,404 @@
...
+          name: labelValue,
+          items: group
+        };
+      })
+      .value();
+  }
+
+  getLabelGroupOptions(items) {
+    const {groupOptions} = this.state;
+    return _.chain(items)
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Avoid `_.chain`... I don't think it works with tree shaking

File: frontend/public/components/overview.jsx [hunk 15]
```diff
@@ -0,0 +1,404 @@
...
+        if (_.has(key, accumulator)) {
+          return accumulator;
+        }
+
+        return {
+          ...accumulator,
+          [key]: key
+        };
+      }, groupOptions)
+      .value();
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Wondering if this is easier if we use `Set`

File: frontend/public/components/overview.jsx [hunk 16]
```diff
@@ -0,0 +1,404 @@
...
+    return _.map(rootResources, rootResource => {
+      const {uid: rootResourceUid} = rootResource.metadata;
+      // Determine the replication controllers associated with this DC
+      const ownedReplicationControllers = this.buildGraphForReplicators(getOwnedResources(replicationControllers.data, rootResourceUid), 'ReplicationController');
+      const ownedReplicaSets = this.buildGraphForReplicators(getOwnedResources(replicaSets.data, rootResourceUid), 'ReplicaSet');
+      const orderedReplicationControllers = this.sortByDeploymentVersion(ownedReplicationControllers, true);
+      const currentController = _.head(orderedReplicationControllers) || _.head(ownedReplicaSets);
+      const prevController = orderedReplicationControllers[1];
+
+      // Get the routes for this deployment config
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: I would take this out for now if we're not showing routes or services yet. (Save the code in a branch :)

File: frontend/public/components/overview.jsx [hunk 17]
```diff
@@ -0,0 +1,404 @@
...
+      const templateLabel = _.get(rootResource, `spec.template.metadata.labels.${kind.toLowerCase()}`);
+      const associatedRoutes = _.reduce(services.data, (accumulator, service) => {
+        const serviceSelector = _.get(service, `spec.selector.${kind.toLowerCase()}`);
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: I'm not sure the label logic is correct. Label matching is pretty complicated.

What is `kind` here?

File: frontend/public/components/overview.jsx [hunk 18]
```diff
@@ -0,0 +1,404 @@
...
+        routes: associatedRoutes
+      };
+    });
+  }
+
+  createOverviewData() {
+    const {deploymentConfigs, deployments, loaded, statefulSets} = this.props;
+    const {selectedGroupLabel} = this.state;
+
+    if (!loaded) {
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: We might look at enabling progressive loading, but fine for now.

File: frontend/public/components/project-overview.jsx [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+      <div className="project-overview__detail project-overview__detail--controller">
+        <ComponentLabel text={_.startCase(currentController.kind)} />
+        <ResourceLink
+          kind={currentController.kind}
+          name={currentController.metadata.name}
+          namespace={namespace}
+        />
+      </div>
+    }
+    { latestVersion &&
```
Comment (spadgett) [inline frontend/public/components/project-overview.jsx]: We could consider leaving this out since it's part of the rc/rs name

File: frontend/public/components/utils/firehose.jsx [hunk 20]
```diff
@@ -174,7 +174,7 @@ export const Firehose = connect(
         if (expand !== this.props.expand) {
           return true;
         }
-        return false;
+        return nextProps.forceUpdate;
```
Comment (spadgett) [inline frontend/public/components/utils/firehose.jsx]: Why is this needed?

File: frontend/public/components/utils/resource-icon.tsx [hunk 21]
```diff
@@ -8,8 +8,7 @@ import { kindToAbbr } from '../../module/k8s/get-resources';
 
 const MEMO = {};
 
-export const ResourceIcon = (props: ResourceIconProps) => {
-  const { kind, className } = props;
+export const ResourceIcon: React.SFC<ResourceIconProps>= ({className, kind}) => {
```
Comment (spadgett) [inline frontend/public/components/utils/resource-icon.tsx]: style nit: space before `=`

File: frontend/yarn.lock [hunk 22]
```diff
@@ -33,17 +33,203 @@
   version "0.7.0"
   resolved "https://registry.yarnpkg.com/@sindresorhus/is/-/is-0.7.0.tgz#9a06f4f137ee84d7df0460c1fdb1135ffa6c50fd"
 
+"@types/c3@^0.6.0":
```
Comment (spadgett) [inline frontend/yarn.lock]: A ton of dependencies are getting added here... Per @jeff-phillips-18, patternfly-react should be doing tree shaking. We should verify we're only loading what we're using.

File: frontend/public/components/utils/firehose.jsx (hunk 20 above)
Comment (TheRealJon) [reply frontend/public/components/utils/firehose.jsx]: Basically, Firehose will only re-render when the underlying resources change. So switching the view component rendered inside the Firehose will not cause a re-render. Since we are taking out the view switching functionality, this probably isn't necessary right now. We could also probably figure out a better way to do this. I just hacked something together to work temporarily.

File: frontend/public/components/project-overview.jsx (hunk 19 above)
Comment (TheRealJon) [reply frontend/public/components/project-overview.jsx]: Yeah, I thought about that too. My only (very minor) concern is that we are leaving it up to the user to know that the version number is in the rc/rs name. But I don't think that's a huge deal because it's pretty obvious.

File: frontend/public/components/overview.jsx (hunk 15 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: This is used in Dropdown. If dropdown accepts a Set or we refactor it to accept a Set, that would work.

File: frontend/public/components/overview.jsx (hunk 17 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: `kind` would be one of 'DeploymentConfig', 'Deployment', or 'StatefulSet' here.

File: frontend/yarn.lock (hunk 22 above)
Comment (spadgett) [reply frontend/yarn.lock]: Best I can tell, all of these dependencies are getting included in the vendor bundle (c3.js, etc.) for overview. It's almost doubling the size from our main vendor bundle. We're only using a couple components. (This is running in development mode. I'll also check the built image.)

@jeff-phillips-18 any ideas?

project overview okd 2018-08-24 09-45-44

project overview okd 2018-08-24 09-42-50

Comment (jeff-phillips-18) [reply frontend/yarn.lock]: Development mode does not tree shake

Comment (spadgett) [reply frontend/yarn.lock]: I was seeing it in the built image, too, but I think I didn't correctly build it. Thanks @jeff-phillips-18, I think we're OK then.

Comment (spadgett) [reply frontend/yarn.lock]: OK guys. Now that I built it correctly, looks like we're good. Sorry for the false alarm!

project overview 2018-08-24 10-13-15

**Comment by spadgett** [comment]
I'm happy we're able to get patternfly-react integrated

File: frontend/yarn.lock (hunk 22 above)
Comment (benjaminapetersen) [reply frontend/yarn.lock]: Curious about that. Is there any concern we will forget about the tree shaking in dev mode & accidentally ship a large bundle to prod in the future?  Or is there any risk that dev mode might load something that won't exist later, meaning we won't see the error now?  Just curious about the possibilities here.

Comment (spadgett) [reply frontend/yarn.lock]: The tree shaking stuff should just work... And with code splitting across routes, it should protect us as well.

Comment (alecmerdler) [reply frontend/yarn.lock]: The "Performance" e2e tests run against the production build, so we are covered from regressions there.

**Comment by TheRealJon** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins push

**Comment by sg00dwin** [comment]
@TheRealJon Here's a screenshot of the empty dropdown filter state that I mentioned.
project-overview

**Comment by coreos-ui** [comment]
The PR image has been pushed to Quay with tag `pr_393_build_417`

File: frontend/public/components/overview.jsx [hunk 23]
```diff
@@ -0,0 +1,353 @@
...
+        groupedItems: this.groupByLabel(filteredItems, this.state.selectedGroupLabel)
+      });
+    } else if (this.state.selectedGroupLabel !== prevState.selectedGroupLabel) {
+      this.setState({
+        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: It's not clear to me why we need to sort by deployment version if we're only showing the active one. We might be able to skip this for now?

File: frontend/public/components/overview.jsx [hunk 24]
```diff
@@ -0,0 +1,353 @@
...
+import { Helmet } from 'react-helmet';
+
+import { StartGuide } from './start-guide';
+import { FLAGS, connectToFlags, flagPending } from '../features';
+import { Dropdown, Firehose, NavTitle, StatusBox } from './utils';
+import { TextFilter } from './factory';
+import { ProjectOverview } from './project-overview';
+
+const getOwnedResources = (resources, uid) => {
+  return _.filter(resources, function (resource) {
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: nit: prefer fat arrow

File: frontend/public/components/overview.jsx [hunk 25]
```diff
@@ -0,0 +1,353 @@
...
+      this.setState({
+        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
+    const compareDeployments = (left, right) => {
+      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
+      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Doesn't this need to be checking the annotations? Looks like the `_.get` call is wrong.

File: frontend/public/components/overview.jsx [hunk 26]
```diff
@@ -0,0 +1,353 @@
...
+    const {deploymentConfigs, deployments, loaded, statefulSets} = this.props;
+    const {selectedGroupLabel} = this.state;
+
+    if (!loaded) {
+      return;
+    }
+
+    const deploymentConfigItems = this.buildGraphForRootResources(deploymentConfigs.data, 'DeploymentConfig');
+    const deploymentItems = this.buildGraphForRootResources(deployments.data, 'Deployment');
+    const statefulSetItems = this.buildGraphForReplicators(statefulSets.data, 'StatefulSet', true);
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: Looks like the method only takes two args?

File: frontend/public/components/project-overview.jsx [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+
+import { ResourceLink } from './utils';
+import { modelFor } from '../module/k8s';
+
+export const ComponentLabel = ({text}) => <div className="co-component-label">{text}</div>;
+
+const ProjectOverviewListItem = ({obj}) => {
+  const {currentController, kind, metadata} = obj;
+  const {namespace, name, uid} = metadata;
+  const kindObj = modelFor(kind);
```
Comment (spadgett) [inline frontend/public/components/project-overview.jsx]: We should ideally be using `connectToModels` here

File: frontend/public/components/project-overview.jsx [hunk 28]
```diff
@@ -0,0 +1,103 @@
...
+      kind={kind}
+      name={name}
+      namespace={namespace}
+    />
+  </h3>;
+
+  const additionalInfo = <div key={uid} className="project-overview__additional-info">
+    { currentController &&
+      <div className="project-overview__detail project-overview__detail--controller">
+        <ComponentLabel text={_.startCase(currentController.kind)} />
```
Comment (spadgett) [inline frontend/public/components/project-overview.jsx]: I wonder if we shouldn't drop the labels in these rows. They seem fairly self-evident.

File: frontend/public/components/project-overview.jsx [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+        <ResourceLink
+          kind={currentController.kind}
+          name={currentController.metadata.name}
+          namespace={namespace}
+        />
+      </div>
+    }
+    <div className="project-overview__detail project-overview__detail--status">
+      <ComponentLabel text="Status" />
+      <Link to={`/k8s/ns/${namespace}/${kindObj.path}/${name}/pods`}>
```
Comment (spadgett) [inline frontend/public/components/project-overview.jsx]: Better to use `resourcePath` or `resourcePathForModel` so we keep the URL building in one place

File: frontend/public/components/overview.jsx (hunk 23 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: My thought was that we will have to use some logic similar to this in order to find the replication controller with the latest version. Since we will eventually be showing the whole list, sorting and grabbing the head seems like a good way to do it to make things easier for showing the whole list. We could make this into something like 'getLatestDeploymentVersion' but we would still have to iterate over all the RC's and do this same logic to return one RC.

File: frontend/public/components/overview.jsx (hunk 26 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: Oops, an artifact left from a previous implementation

File: frontend/public/components/overview.jsx (hunk 23 above)
Comment (spadgett) [reply frontend/public/components/overview.jsx]: You should be able to look at the property on the deployment config or deployment to get the current revision. rc-10 might be failed and rc-9 is what is running... It's not necessarily the most recent by number

Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: Gotcha, okay, that makes sense

File: frontend/public/components/overview.jsx [hunk 30]
```diff
@@ -68,33 +70,38 @@ class OverviewDetails extends React.Component {
...
     }
   }
 
-  sortByDeploymentVersion(replicationControllers, descending) {
-    const compareDeployments = (left, right) => {
-      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
-      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
+  sortReplicatorsByVersion(replicators, kind, descending) {
+    const compare = (left, right) => {
+      const annotation = kind === 'ReplicationController' ? 'openshift.io/deployment-config.latest-version' : 'deployment.kubernetes.io/revision';
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: I think it would be a little cleaner not to pass kind here, but to instead pass either the annotation name or a callback for getting the revision. Something like

```js
sortByRevision(replicators, descending, annotation) ...

sortRC(replicationControllers, descending) => sortByRevision(replicationControllers, descending, 'openshift.io/deployment-config.latest-version');

sortRS(replicaSets, descending) => sortByRevision(replicaSets, descending, 'deployment.kubernetes.io/revision');
```

File: frontend/public/components/overview.jsx [hunk 31]
```diff
@@ -135,24 +142,37 @@ class OverviewDetails extends React.Component {
     });
   }
 
-  groupByLabel(items, label) {
-    const groups = _.groupBy(items, (item) => _.get(item, ['metadata', 'labels', label], 'other'));
+  groupItems(items, label) {
+    const compareGroups = (a, b) => {
+      if (a.name === 'other') {
```
Comment (spadgett) [inline frontend/public/components/overview.jsx]: This could potentially collide with a value since "other" is a legal label value

File: frontend/public/components/overview.jsx [hunk 32]
```diff
@@ -0,0 +1,372 @@
...
+}
+
+OverviewPage_.displayName = 'OverviewPage';
+
+OverviewPage_.propTypes = {
+  match: PropTypes.object.isRequired,
+  flags: PropTypes.object.isRequired
+};
+
+export const OverviewPage = connectToFlags(FLAGS.OPENSHIFT, FLAGS.PROJECTS_AVAILABLE)(OverviewPage_);
```
[the remark this answers is not in the document]
Comment (spadgett) [reply frontend/public/components/overview.jsx]: One drawback to this is you can't have private components. You have to export them all. It becomes more difficult to tell what's used internally and what's meant to be reused. It also discourages breaking things into small, simple components because it's a lot more work.

This would be a large refactoring effort and make it harder to backport fixes to previous releases as well because everything has moved. If you navigate to symbols instead of files in your editor, I feel like it shouldn't matter much what file the component is in.

File: frontend/public/components/overview.jsx (hunk 31 above)
Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: This issue goes beyond just this sorting function. In the case where a resource has a label value of 'other', it would be grouped with resources which don't have a value for the currently selected 'group by' label. Since labels can contain almost any string value, isn't the possibility of collision always going to be an issue?

Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: Yeah, that would work to group them into a guaranteed unique group, but we would still have a problem with what to display as the heading for that group since we don't want that to collide either. We could use a value with a space in it which would make it an invalid label value, and it could still be used as a display value so that we don't have to do special logic when rendering. "No value" or something like that.

Comment (spadgett) [reply frontend/public/components/overview.jsx]: Let's look at it in a follow on

Comment (TheRealJon) [reply frontend/public/components/overview.jsx]: https://jira.coreos.com/browse/CONSOLE-755

File: frontend/public/components/project-overview.jsx (hunk 28 above)
Comment (TheRealJon) [reply frontend/public/components/project-overview.jsx]: Follow on?
